lib/private/Settings/DeclarativeManager.php

```diff
@@ -290,18 +292,54 @@ private function getForm(string $app, string $formId): ?IDeclarativeSettingsForm
 	private function getInternalValue(IUser $user, string $app, string $formId, string $fieldId): mixed {
 		$sectionType = $this->getSectionType($app, $fieldId);
 		$defaultValue = $this->getDefaultValue($app, $formId, $fieldId);
+
+		$form = $this->getForm($app, $fieldId);
+		if ($form === null) {
+			$field = $this->getSchemaField($app, $formId, $fieldId);
+		} else {
+			$field = $this->getSchemaField($app, $formId, $fieldId, $form);
+		}
+		$isSensitive = $field !== null && isset($field['sensitive']) && $field['sensitive'];
```

Comment:
```suggestion
		$isSensitive = $field !== null && isset($field['sensitive']) && $field['sensitive'] === true;
```
Again, strict check to avoid matching something else non-falsy.